beets/autotag/__init__.py

```diff
@@ -154,9 +154,14 @@ def apply_metadata(album_info, mapping):
                       'albumdisambig',
                       'releasegroupdisambig',
                       'data_source',):
+            # Don't overwrite fields with empty values unless the
+            # field is explicitly allowed to be overwritten
+            clobber = field not in config['no_clobber'].get()
```

Comment:
A type assertion here might be useful: `.as_str_seq()` instead of `.get()`.

Reply:
Good catch. Fixed.